simple-facebook-instant-articles.php

```diff
@@ -405,10 +406,32 @@ public function api_galleries_shortcode( $atts ) {
 	 * @return string           FB IA formatted markup for social embeds.
 	 */
 	public function reformat_social_embed( $html, $url, $attr, $post_ID = null ) {
+
 		return sprintf( '<figure class="op-social"><iframe>%s</iframe></figure>', $html );
 	}
 
 	/**
+	 * Ensure facebook scripts are loaded for facebook embeds.
+	 *
+	 * @param string   $html    HTML markup to be embeded into post content.
+	 * @param string   $url     The attempted embed URL.
+	 * @param array    $attr    An array of shortcode attributes.
+	 *
+	 * @return string           Facebook embed code with required script.
+	 */
+	public function load_facebook_scripts( $html, $url, $attr ) {
+
+		// If the embed is any kind of facebook embed, try and load the Facebook SDK.
+		// Can't use precise regex, as we don't really know what WP.com is doing here!
+		if ( false !== strpos( $url, 'facebook.com' ) ) {
+
+			$html .= '<div id="fb-root"></div> <script>(function(d, s, id) { var js, fjs = d.getElementsByTagName(s)[0]; if (d.getElementById(id)) return; js = d.createElement(s); js.id = id; js.src = "//connect.facebook.net/en_US/all.js#xfbml=1"; fjs.parentNode.insertBefore(js, fjs); }(document, "script", "facebook-jssdk"));</script>';
```

Comment:
Should we check to make sure we're only adding this once? Otherwise multiple FB embeds in a post will load the SDK multiple times.